VisualStudioPlugin/SolutionExplorerMenuCommand.cs

```diff
@@ -239,7 +258,7 @@ private bool UploadFilesToServer(int selectedProjectId, IEnumerable<SelectedItem
 
             while (compileStatus.State == Api.CompileState.InQueue)
             {
-                Thread.Sleep(2000);
+                System.Threading.Tasks.Task.Delay(2000);
```

Comment:
Since this task is not awaited, this method will keep on going. I'd be interested to see how many times we invoke `api.ReadCompile`  -- typically async-ness infects the entire call stack (or until someone does a `Task.Run` or similar).

Reply:
Your right! the method kept on going 😅, it queried a couple of times (in debug).
I've changed it to work with the same async logic as the backtest method does 👍 . With the change, in debug it's also queering twice.

Reply:
Very cool, thanks for updating. I think this is ready for merge -- we may want to browse the vs plugin project for any other usages of `Thread.Sleep` and see if w can unroll them into `async/await` like we've done here.  If it becomes common, an `async` version of the API methods might be useful and would remove all the `Task.Run` calls and make it *even faster* :)

Reply:
I've doubled checked and there are no other `Thread.Sleep` in the plugin